x/vendorinfo/types/messages_vendor_info_test.go

```diff
@@ -1,104 +1,303 @@
+				CompanyLegalName: testconstants.CompanyLegalName,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "vid is 0",
+			msg: MsgCreateVendorInfo{
+				Creator:          sample.AccAddress(),
+				VendorID:         0,
+				VendorName:       testconstants.VendorName,
+				CompanyLegalName: testconstants.CompanyLegalName,
+			},
+			err: validator.ErrFieldLowerBoundViolated,
+		},
+		{
+			name: "vid is bigger than 65535",
+			msg: MsgCreateVendorInfo{
+				Creator:          sample.AccAddress(),
+				VendorID:         65536,
+				VendorName:       testconstants.VendorName,
+				CompanyLegalName: testconstants.CompanyLegalName,
+			},
+			err: validator.ErrFieldUpperBoundViolated,
+		},
+		{
+			name: "vendor name len < 2",
+			msg: MsgCreateVendorInfo{
+				Creator:          sample.AccAddress(),
+				VendorID:         testconstants.VendorID1,
+				VendorName:       "a",
+				CompanyLegalName: testconstants.CompanyLegalName,
+			},
+			err: validator.ErrFieldMinLengthNotReached,
+		},
+		{
+			name: "vendor name len > 32",
+			msg: MsgCreateVendorInfo{
+				Creator:          sample.AccAddress(),
+				VendorID:         testconstants.VendorID1,
+				VendorName:       "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789",
+				CompanyLegalName: testconstants.CompanyLegalName,
+			},
+			err: validator.ErrFieldMaxLengthExceeded,
+		},
+		{
+			name: "company legal name len < 2",
+			msg: MsgCreateVendorInfo{
+				Creator:          sample.AccAddress(),
+				VendorID:         testconstants.VendorID1,
+				VendorName:       testconstants.VendorName,
+				CompanyLegalName: "a",
+			},
+			err: validator.ErrFieldMinLengthNotReached,
+		},
+		{
+			name: "company legal name len > 64",
+			msg: MsgCreateVendorInfo{
+				Creator:          sample.AccAddress(),
+				VendorID:         testconstants.VendorID1,
+				VendorName:       testconstants.VendorName,
+				CompanyLegalName: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789123",
+			},
+			err: validator.ErrFieldMaxLengthExceeded,
+		},
+		{
+			name: "company preffered name len > 64",
+			msg: MsgCreateVendorInfo{
+				Creator:              sample.AccAddress(),
+				VendorID:             testconstants.VendorID1,
+				VendorName:           testconstants.VendorName,
+				CompanyLegalName:     testconstants.CompanyPreferredName,
+				CompanyPrefferedName: "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789123",
+			},
+			err: validator.ErrFieldMaxLengthExceeded,
+		},
+		{
+			name: "vendor landing page URL is not URL",
+			msg: MsgCreateVendorInfo{
+				Creator:              sample.AccAddress(),
+				VendorID:             testconstants.VendorID1,
+				VendorName:           testconstants.VendorName,
+				CompanyLegalName:     testconstants.CompanyPreferredName,
+				VendorLandingPageURL: "ABC",
+			},
+			err: validator.ErrFieldNotValid,
+		},
+		{
+			name: "vendor landing page URL len > 256",
+			msg: MsgCreateVendorInfo{
```

Comment:
Makes sense to add a couple of more negative tests:
1) VendorName is is not set (empty string). Expected: ErrRequiredFieldMissing
2) CompanyLegalName is is not set (empty string). Expected: ErrRequiredFieldMissing